File: codegen/results/v001/qwen3.5_4b_think/20260408_204208/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

// Read price from command-line argument, default to 100
const price = Number(process.argv[2] ?? 100);

// Define test cases
const tests = [
    { 
        description: "quantity 9 (Full Price)", 
        quantity: 9, 
        expected: price * 9 
    },
    { 
        description: "quantity 10 (90% Discount)", 
        quantity: 10, 
        expected: price * 10 * 0.9 
    },
    { 
        description: "quantity 49 (90% Discount)", 
        quantity: 49, 
        expected: price * 49 * 0.9 
    },
    { 
        description: "quantity 50 (75% Discount)", 
        quantity: 50, 
        expected: price * 50 * 0.75 
    },
];

// Run tests and output results
for (const test of tests) {
    const result = calculateDiscount(price, test.quantity);
    // Use strict equality check with a tiny tolerance for floating point safety
    if (Math.abs(result - test.expected) < 0.0001) {
        console.log(`PASS: ${test.description}`);
    } else {
        console.log(`FAIL: ${test.description} (expected: ${test.expected}, got: ${result})`);
    }
}

File: codegen/results/v001/qwen3.5_4b_think/20260408_204208/go/005_unit_test_writer/solution.go
package main

import (
	"fmt"
	"math"
)

func main() {
	price := 100.0

	// Test case 1: quantity = 9
	desc := "quantity = 9"
	expected := price * 9.0
	got := calculateDiscount(price, 9)
	test(desc, expected, got)

	// Test case 2: quantity = 10
	desc = "quantity = 10"
	expected = price * 10.0 * 0.9
	got = calculateDiscount(price, 10)
	test(desc, expected, got)

	// Test case 3: quantity = 49
	desc = "quantity = 49"
	expected = price * 49.0 * 0.9
	got = calculateDiscount(price, 49)
	test(desc, expected, got)

	// Test case 4: quantity = 50
	desc = "quantity = 50"
	expected = price * 50.0 * 0.75
	got = calculateDiscount(price, 50)
	test(desc, expected, got)
}

func test(desc string, expected float64, got float64) {
	// Use a small epsilon for floating point comparison
	if math.Abs(got - expected) < 0.00001 {
		fmt.Println("PASS:", desc)
	} else {
		fmt.Println("FAIL:", desc, "(expected:", expected, ", got:", got, ")")
	}
}

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}